api/app/signals/apps/signals/managers.py

```diff
@@ -165,6 +171,28 @@ def split(self, split_data, signal):
                 transaction.on_commit(lambda: create_child.send(sender=self.__class__,
                                                                 signal_obj=child_signal))
 
+                # Check if we need to copy the images of the parent
+                if 'reuse_parent_image' in validated_data and validated_data['reuse_parent_image']:
+                    parent_image_qs = parent_signal.attachments.filter(is_image=True)
+                    if parent_image_qs.exists():
+                        for parent_image in parent_image_qs.all():
+                            # Copy the original file and rename it by pre-pending the name with
+                            # split_{loop_counter}_{original_name}
+                            try:
+                                attachment = Attachment()
+                                attachment._signal = child_signal
+
+                                child_image_name = 'split_{}_{}'.format(
+                                    loop_counter,
+                                    parent_image.file.name.split('/').pop()
+                                )
+                                attachment.file.save(name=child_image_name,
```

Comment:
I guess this is the one line that may actually raise a `FileNotFoundError`. You can make the `try` /  `except` block more narrow.

 Consider moving `attachment.save()` to `else:` clause, and moving the `attachment` definition to the before  the `try` clause as well.